doc/src/manual/conversion-and-promotion.md

```diff
@@ -134,52 +125,103 @@ actual Julia behaviour. This is the actual implementation in Julia:
 ```
 
 ### [Case Study: Rational Conversions](@id man-rational-conversion)
 
-To continue our case study of Julia's `Rational` type, here are the conversions declared in [rational.jl](https://github.com/JuliaLang/julia/blob/master/base/rational.jl),
+To continue our case study of Julia's `Rational` type, here are conversions analogous
+to those declared in [rational.jl](https://github.com/JuliaLang/julia/blob/master/base/rational.jl),
 right after the declaration of the type and its constructors:
 
-```julia
-convert{T<:Integer}(::Type{Rational{T}}, x::Rational) = Rational(convert(T,x.num),convert(T,x.den))
-convert{T<:Integer}(::Type{Rational{T}}, x::Integer) = Rational(convert(T,x), convert(T,1))
-
-function convert{T<:Integer}(::Type{Rational{T}}, x::AbstractFloat, tol::Real)
-    if isnan(x); return zero(T)//zero(T); end
-    if isinf(x); return sign(x)//zero(T); end
-    y = x
-    a = d = one(T)
-    b = c = zero(T)
-    while true
-        f = convert(T,round(y)); y -= f
-        a, b, c, d = f*a+c, f*b+d, a, b
-        if y == 0 || abs(a/b-x) <= tol
-            return a//b
-        end
-        y = 1/y
+```@meta
+DocTestSetup = quote
+    struct OurRational{T<:Integer} <: Real
+               num::T
+               den::T
+               function OurRational{T}(num::T, den::T) where T<:Integer
+                   if num == 0 && den == 0
+                        error("invalid rational: 0//0")
+                   end
+                   g = gcd(den, num)
+                   num = div(num, g)
+                   den = div(den, g)
+                   new(num, den)
+               end
+           end
+    OurRational(n::T, d::T) where T<:Integer = OurRational{T}(n,d)
+    OurRational(n::Integer) = OurRational(n,one(n))
+    //(n::Integer, d::Integer) = OurRational(n,d)
+    //(x::OurRational, y::Integer) = x.num // (x.den*y)
+    //(x::Integer, y::OurRational) = (x*y.den) // y.num
+    //(x::Complex, y::Real) = complex(real(x)//y, imag(x)//y)
+    //(x::Real, y::Complex) = x*y'//real(y*y')
+    function //(x::Complex, y::Complex)
+        xy = x*y'
+        yy = real(y*y')
+        complex(real(xy)//yy, imag(xy)//yy)
     end
+    promote_rule{T<:Integer,S<:Integer}(::Type{OurRational{T}}, ::Type{S}) = OurRational{promote_type(T,S)}
+    promote_rule{T<:Integer,S<:Integer}(::Type{OurRational{T}}, ::Type{OurRational{S}}) = OurRational{promote_type(T,S)}
+    promote_rule{T<:Integer,S<:AbstractFloat}(::Type{OurRational{T}}, ::Type{S}) = promote_type(T,S)
+```
 end
-convert{T<:Integer}(rt::Type{Rational{T}}, x::AbstractFloat) = convert(rt,x,eps(x))
+```
 
-convert{T<:AbstractFloat}(::Type{T}, x::Rational) = convert(T,x.num)/convert(T,x.den)
-convert{T<:Integer}(::Type{T}, x::Rational) = div(convert(T,x.num),convert(T,x.den))
+```jldoctest rational
+julia> convert{T<:Integer}(::Type{OurRational{T}}, x::OurRational) = OurRational(convert(T,x.num),convert(T,x.den))
+convert (generic function with 1 method)
```

Comment:
this should be importing Base.convert - anywhere a doctest says "generic function with 1 method" for a base function, it's not showing what it should be. there were a few cases of this already merged that should also be cleaned up